test/test_testing.py

```diff
@@ -258,6 +299,80 @@ def cleanup():
         # The charm_dir also gets set
         self.assertEqual(harness.framework.charm_dir, tmp)
 
+    def test_relation_set_deletes(self):
+        # language=YAML
+        harness = Harness(CharmBase, meta='''
+            name: test-charm
+            requires:
+                db:
+                    interface: pgsql
+            ''')
+        harness.begin()
+        harness.set_leader(False)
+        rel_id = harness.add_relation('db', 'postgresql')
+        harness.update_relation_data(rel_id, 'test-charm/0', {'foo': 'bar'})
+        harness.add_relation_unit(rel_id, 'postgresql/0')
+        rel = harness.charm.model.get_relation('db', rel_id)
+        del rel.data[harness.charm.model.unit]['foo']
+        self.assertEqual({}, harness.get_relation_data(rel_id, 'test-charm/0'))
+
+    def test_get_backend_calls(self):
+        # language=YAML
+        harness = Harness(CharmBase, meta='''
+            name: test-charm
+            requires:
+                db:
+                    interface: pgsql
+            ''')
+        harness.begin()
+        # No calls to the backend yet
+        self.assertEqual([], harness.get_backend_calls())
+        rel_id = harness.add_relation('db', 'postgresql')
+        # update_relation_data ensures the cached data for the relation is wiped
+        harness.update_relation_data(rel_id, 'test-charm/0', {'foo': 'bar'})
+        self.assertEqual([
+            ('relation_ids', 'db'),
+            ('relation_list', rel_id),
+        ], harness.get_backend_calls(reset=True))
```

Comment:
This is the awkward bit.. what just happened there?  How and why do we constrain the above logic to that list?  Why does it matter?